src/main/java/org/reactome/orthoinference/PathwaysInferrer.java

```diff
@@ -78,17 +78,10 @@ private static void createInferredPathwayHierarchy(GKInstance sourceEventInst) t
 
 		for (GKInstance sourcePathwayReferralInst : sourcePathwayReferralInstances)
 		{
-			// Disease TopLevelPathway inference is skipped.
-			if (sourcePathwayReferralInst.equals(diseasePathwayInst)) {
-				logger.info("Pathway referral is " + diseasePathwayInst + " -- skipping Pathway inference");
-				return;
-			}
 
 			logger.info("Generating inferred Pathway: " + sourcePathwayReferralInst);
 			// Pathways that have been inferred already are skipped, as are Pathways that are only children of the Disease TopLevelPathway.
-			if (alreadyInferred(sourcePathwayReferralInst) || InstanceUtilities.onlyInDiseasePathway(sourcePathwayReferralInst)) {
-				logger.info("Inferred pathway instance {} already exists or only has Disease for a top level pathway", sourcePathwayReferralInst.getExtendedDisplayName());
-			} else {
+			if (sourceInstanceToInferredInstance.get(sourcePathwayReferralInst) == null && !InstanceUtilities.onlyInDiseasePathway(sourcePathwayReferralInst)) {
```

Comment:
Maybe the first half of the condition can be wrapped in a descriptive method name like `noInferenceFor(sourcePathwayReferralInst)`

Reply:
I used `hasNotBeenInferred(sourcePathwayReferralInst)`